backend/controllers/user.controller.js

```diff
@@ -1,37 +1,70 @@
 const user = require('../models/user.model.js');
+const jwt = require('jsonwebtoken');
+require('dotenv').config();
+const bcrypt = require('bcrypt');
 
 const createUser = async (req,res)=>{
     try {
         const {username, password, email} = req.body;
     if (!username || !email || !password) {
       return  res.status(401).json({message: 'All fields are required'});
     };
-        const User = await user.create(req.body);
-        res.status(200).json(User)
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(password, salt);
+
+
+        const User = await user.create({
+          username,
+          email,
+          password: hashedPassword,
+        });
+        res.status(200).json(User);
+      
+   
+
     } catch (error) {
         res.status(500).json({message: "internal error"})
     };
  };
 
+
  const loginUser = async (req, res) => {
-     try {
-       const { username, password } = req.body;
-       const existingUser = await user.findOne({ username });
-       if (!existingUser) {
-         return res.status(401).json({ message: 'Invalid username' });
-       }
-   
-       const isMatch = await existingUser.matchPassword(password);
-       if (!isMatch) {
-         return res.status(401).json({ message: 'Invalid password' });
-       }
+  try {
+    const { email, password } = req.body;
+
+    const existingUser = await user.findOne({ email });
+    if (!existingUser) {
+      return res.status(401).json({ message: 'Invalid email' });
```

Comment:
For security reasons, it's best not to specify which credential is incorrect when showing an error messageogin for l. Instead, use a generic response like Invalid email or password" This approach helps prevent attackers from determining valid usernames or emails through error messages.